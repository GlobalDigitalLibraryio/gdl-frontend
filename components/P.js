// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 * 
 * See LICENSE
 */

import styled from 'styled-components';
import { fontSize } from 'styled-system';
import lineHeight from './helpers/lineHeight';

const P = styled.p`
  ${fontSize};
  ${lineHeight};
`;

export default P;
