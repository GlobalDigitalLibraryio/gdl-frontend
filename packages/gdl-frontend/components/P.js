// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import styled from 'react-emotion';
import { fontSize, color } from 'styled-system';
import lineHeight from '../style/lineHeight';

const P = styled('p')`
  ${fontSize};
  ${lineHeight};
  ${color};
  &:last-child {
    margin-bottom: 0;
  }
`;

export default P;
