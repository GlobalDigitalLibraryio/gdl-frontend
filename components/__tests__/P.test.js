// @flow
/**
 * Part of GDL gdl-frontend.
 * Copyright (C) 2017 GDL
 *
 * See LICENSE
 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import P from '../P';

test('Responsive font sizes', () => {
  const tree = shallow(<P fontSize={[15, 20]}>Responsive font size</P>);

  expect(toJson(tree)).toMatchSnapshot();
});

test('Responsive line heights', () => {
  const tree = shallow(<P lineHeight={[15, 20]}>Responsive font size</P>);

  expect(toJson(tree)).toMatchSnapshot();
});
