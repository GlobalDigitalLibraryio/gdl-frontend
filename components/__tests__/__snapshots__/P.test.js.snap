// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive font sizes 1`] = `
.emotion-0 {
  font-size: 15px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-0:last-child {
  margin-bottom: 0;
}

<p
  className="emotion-0 emotion-1"
  fontSize={
    Array [
      15,
      20,
    ]
  }
>
  Responsive font size
</p>
`;

exports[`Responsive line heights 1`] = `
.emotion-0 {
  line-height: 15px;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    line-height: 20px;
  }
}

.emotion-0:last-child {
  margin-bottom: 0;
}

<p
  className="emotion-0 emotion-1"
>
  Responsive font size
</p>
`;
